src/library/scala/collection/Iterable.scala

```diff
@@ -194,7 +194,19 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     *           applied to this $coll. By default the string prefix is the
     *           simple name of the collection class $coll.
     */
-  def className: String = {
+  @`inline` protected[this] def className: String = stringPrefix
+
+  /** Forwarder to `className` for use in `scala.runtime.ScalaRunTime`.
+    * 
+    * This allows the proper visibility for `className` to be
+    * published, but provides the exclusive access needed by
+    * `scala.runtime.ScalaRunTime.stringOf` (and a few tests in
+    * the test suite).
+    */
+  private[scala] def collectionClassName: String = className
```

Comment:
This should probably be `final`

Reply:
Yes, it probably should and could be.  I've made it so.